freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UsersState.java

```diff
@@ -67,6 +67,14 @@ public Builder addMemberToGroup(String group, String user) {
             return this;
         }
 
+        public Builder addMemberToGroupIfGroupExists(String group, String user) {
+            boolean groupExists = fmsGroups.stream().anyMatch(g -> g.getName().equals(group));
```

Comment:
I like the idea. It looks like even if the group is deleted, the associations are not deleted in IAM. So the users' group association keeps growing in IAM. This fix tries to reduce the group footprint in FreeIPA by not adding such groups.

Codewise, is it possible to create a HashSet for fmsGroups' name?